docs: update documentation to use `ng generate config` and `ng generate environments` (#48757)
angular/angular: aio/content/guide/build.md

## Configuring application environments

> You can define different named build configurations for your project, such as _development_ and _staging_, with different defaults.

Reviewer: A couple of things:
* The`*` for italics used instead of the underscore in AIO (with only a few exceptions that haven't been corrected, yet).
* Should these terms be formatted as code since they're used in filenames?
```suggestion
You can define different named build configurations for your project, such as `development` and `staging`, with different defaults.
```

Each named configuration can have defaults for any of the options that apply to the various [builder targets](guide/glossary#target), such as `build`, `serve`, and `test`.
The [Angular CLI](cli) `build`, `serve`, and `test` commands can then replace files with appropriate versions for your intended target environment.

### Configure environment-specific defaults

> Using the Angular CLI, start by running the below [generate environments command](cli/generate#environments-command) to create the `src/environments/` directory and configuring the application project to use these files.

Reviewer: ```suggestion
Using the Angular CLI, start by running the [generate environments command](cli/generate#environments-command) shown here to create the `src/environments/` directory and configure the project to use these files.
```

<code-example format="shell" language="shell">

ng generate environments

</code-example>

> The project's `src/environments/` directory contains the base configuration file, `environment.ts`, which provides configuration for the default environment which is production.

Reviewer: ```suggestion
The project's `src/environments/` directory contains the base configuration file, `environment.ts`, which provides configuration for `production`, the default environment.
```

> You can add override defaults for additional environments, such as development and staging, in target-specific configuration files.

Reviewer: ```suggestion
You can override default values for additional environments, such as `development` and `staging`, in target-specific configuration files.
```

<code-example format="typescript" language="typescript">

export const environment = {
  production: true,
  apiUrl: 'http://my-prod-url'
};

</code-example>

You can add target-specific configuration files, such as `environment.development.ts`.
The following content sets default values for the development build target:

<code-example format="typescript" language="typescript">

export const environment = {
  production: false,
  apiUrl: 'http://my-api-url'
};

</code-example>

> ### Using environment-specific variables in your application

Reviewer: Starting in 2023, we're using `app`.

```suggestion
### Using environment-specific variables in your app
```

Author: Mind if I ask the reasoning behind this decision? 

Was under the impression that we favoured application and updated portions of our docs from app to application.

Reviewer: We did favor `application` for a long time, but while we were updating the landing page and introductory content in Dec. 2022, it became apparent that it sounded a bit clunky and out-of-date.

My understanding is that `application` was preferred over `app` for a long time to avoid confusion with mobile apps, but apparently that's not as big of an issue, these days. So, there's no mandate to globally replace `application`, just to use `app` in new content and update `application` as existing content is reviewed and updated.

Author: Thanks for the explanation

> The following application structure configures build targets for development and staging environments:

Reviewer: ```suggestion
The following application structure configures build targets for `development` and `staging` environments:
```

  Firefox ESR
  </code-example>

> To override the internal configuration use the [`ng generate browserslist`](cli/generate#config-command) command to generate a `.browserslistrc` configuration file in the the project directory.

Reviewer: ```suggestion
To override the internal configuration, run [`ng generate browserslist`](cli/generate#config-command), which generates a `.browserslistrc` configuration file in the the project directory.
```
